app/domain/authentication/authn_jwt/input_validation/validate_uri_based_parameters.rb

```diff
@@ -0,0 +1,55 @@
+module Authentication
+  module AuthnJwt
+    module InputValidation
+      ValidateUriBasedParameters ||= CommandClass.new(
+        dependencies: {
+          # ValidateWebserviceIsWhitelisted calls ValidateAccountExists
+          # we call ValidateAccountExists for better readability and safety
+          validate_account_exists: ::Authentication::Security::ValidateAccountExists.new,
+          validate_webservice_is_whitelisted: Security::ValidateWebserviceIsWhitelisted.new
+        },
+        inputs: %i[authenticator_input enabled_authenticators]
+      ) do
+        # authenticator_input attributes
+        #  :account - from uri - has explicit validation
+        #  :authenticator_name - from uri - has implicit validation with service_id
+        #  :service_id - from uri - has implicit validation with authenticator_name
+        #  :username - can be from uri
+        #    - validation is specific for authenticator
+        #    - in jwt case undefined until token signature validation
+        #  :credentials - the web request body - validation is specific for authenticator
+        #  :client_ip - from metadata or request - depends username
+        #  :request - the web request object
```

Comment:
i'd rather this would written in the `authenticator_input` in case it changes. no one will find this class to change this comment

Reply:
Removed the comment